src/transformStateWithClones.js

```diff
@@ -7,7 +7,34 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const history = [];
+  let newState = { ...state };
+
+  for (const action of actions) {
+    let modifiedState = { ...newState };
+
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(modifiedState, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete modifiedState[key];
+        }
+        break;
+
+      case 'clear':
+        modifiedState = {};
+        break;
+      default:
+        throw new Error('Error');
```

Comment:
The error message 'Error' in the default case is not descriptive. Consider providing a more informative error message, such as 'Unknown action type' or including the action type in the message for better debugging.

Reply:
> The error message 'Error' in the default case is not descriptive. Consider providing a more informative error message, such as 'Unknown action type' or including the action type in the message for better debugging.

Good point